packages/builders/src/jest/jest.builder.ts

```diff
@@ -17,27 +17,47 @@ try {
 const { runCLI } = require('jest');
 
 export interface JestBuilderOptions {
+  codeCoverage?: boolean;
   jestConfig: string;
+  testDirectory?: string;
+  testFile?: string;
+  setupFile?: string;
   tsConfig: string;
-  watch: boolean;
-  bail?: boolean;
+  bail?: number | boolean;
   ci?: boolean;
-  codeCoverage?: boolean;
-  onlyChanged?: boolean;
+  color?: boolean;
+  json?: boolean;
   maxWorkers?: number;
+  onlyChanged?: boolean;
+  outputFile?: string;
   passWithNoTests?: boolean;
   runInBand?: boolean;
-  setupFile?: string;
   silent?: boolean;
-  updateSnapshot?: boolean;
   testNamePattern?: string;
+  updateSnapshot?: boolean;
+  useStderr?: boolean;
+  watch?: boolean;
+  watchAll?: boolean;
 }
 
 export default class JestBuilder implements Builder<JestBuilderOptions> {
   run(
     builderConfig: BuilderConfiguration<JestBuilderOptions>
   ): Observable<BuildEvent> {
     const options = builderConfig.options;
+
+    if (options.testFile) {
+      const filePath = options.testDirectory
+        ? path.resolve(options.testDirectory, options.testFile)
+        : realpathSync(options.testFile);
+
+      if (filePath && !this._isInFolder(builderConfig.root, filePath)) {
+        return of({
+          success: true
```

Comment:
This seems like an error to me. If we look at this out of the context of `vscode-jest` then running the following as a command will pass without errors.

```
> ng test app1 --test-file apps/app2/src/app/app.component.spec.ts
Tests Passed?
```

For VSCode-Jest debugging, it's hard to get the project which is being tested, thus you are testing all of them with your config. Let's consider the following solutions:

1. Compute which project needs to be tested (i don't think VSCode debugging has support for this :()
2. Pass the `testFile` to all projects like you are doing now but also pass `passWithNoTests` so when it doesn't match any files, it will still pass 🤷‍♀️. This solves your use case semantically without causing a false positive for others.
3. Prompt the user for the Project via https://code.visualstudio.com/docs/editor/variables-reference#_input-variables
4. Add configuration to https://github.com/jest-community/vscode-jest to adjust it's logic if the user is using `@nrwl/builders:jest` similar to how it has specific treatment for `create-react-app` projects.

Reply:
@FrozenPandaz Jason, my intent for --test-file was rather specific to vscode-jest integration (or a similar scenario). I did not want to start testing other projects than the one holding the test-file. This way the terminal output is clear. I have also added --pass-with-no-tests to launch.json (see my edited commend above).
Using ng test with --test-file and non-matching project name is not a supported scenario and produces no results (as expected). The hole --test-directory affair is also very much VSCode specific, being the easiest way to provide full path to the file when needed and ignore it when --test-file is empty.